include/revng/ABI/FunctionType/Layout.h

```diff
@@ -222,4 +222,30 @@ inline uint64_t finalStackOffset(const model::TypePath &Function) {
   return finalStackOffset(*Function.get());
 }
 
+struct UsedRegisters {
+  llvm::SmallVector<model::Register::Values, 8> Arguments;
+  llvm::SmallVector<model::Register::Values, 8> ReturnValues;
```

Comment:
`DenseSet`?

Reply:
Wait, no, it has to be a vector - order matters.

Now it returns them ordered the same way `sortArguments`/`sortReturnValues` returns (as in, order depending on the ABI definition), switching to a set would mean ordering them the same way `model::Register::Values` does.